File: orderAhead-BE/models/datatable_strain.py
from models.datatable import Datatable

class DatatableStrain(Datatable):
  def build_sql(self):
    sql = 'SELECT "Strain Name", SUM("Weight Sold") FROM "Sales_by_item" GROUP BY "Strain Name";'
    return sql

  def field_names(self):
    return ['strain', 'lbs_sold']

File: orderAhead-FE/src/components/datatable/DatatableStrain.js
import React, { useEffect, useMemo, useState } from 'react'
import Datatable from '../Datatable'
import { dashboardServices } from '../../controllers/_services/dashboard.service'

const DatatableStrain = (props) => {

  const hearderColumns = useMemo(() => {
    return [
      {
        label: 'Strain Name',
        field: 'strain',
      },
      {
        label: 'lbs. sold',
        field: 'lbs_sold',
      },
    ]
  })

  const [datatable, setDatatable] = useState({
    columns: hearderColumns,
    rows: [],
  })
  useEffect(() => {
    dashboardServices.loadDatatableStrain().then((response) => {
      setDatatable({
        columns: hearderColumns,
        rows: response.data
      })
    })
  }, [])

  return (
    <>
      <Datatable data={datatable} />
    </>
  )
}

export default DatatableStrain;